os/drivers/video/video.c

```diff
@@ -518,13 +527,22 @@ static int video_qbuf(FAR video_upperhalf_t *priv, FAR struct v4l2_buffer *buf)
 	if (buf->type == V4L2_BUF_TYPE_VIDEO_CAPTURE) {
 		video_lock(&priv->still_inf.lock_state);
 		next_video_state = estimate_next_video_state(priv, CAUSE_VIDEO_START);
-		change_video_state(priv, next_video_state);
+		ret = change_video_state(priv, next_video_state);
 		video_unlock(&priv->still_inf.lock_state);
+		if (ret < 0) {
+			return ret;
+		}
 	} else {
 		container = video_framebuff_get_dma_container(&type_inf->bufinf);
 		if (container) {
-			video_devops->set_buftype(priv->dev, buf->type);
-			video_devops->set_buf(priv->dev, container->buf.m.userptr, container->buf.length);
+			ret = video_devops->set_buftype(priv->dev, buf->type);
```

Comment:
video_unlock is unncessary? (it locked in line 519)

Reply:
Video lock is taken when changing states, and same is unlocked at line number 522.

Reply:
if state is VIDEO_STATE_STREAMON, type_inf->lock_state is still locked here, and it returns ret.

Reply:
Corrected!